traffic_ops/traffic_ops_golang/topology/topologies.go

```diff
@@ -150,6 +150,16 @@ func (topology *TOTopology) Validate() error {
 	rules["empty cachegroups"] = topology_validation.CheckForEmptyCacheGroups(topology.ReqInfo.Tx, cacheGroupIds, dsCDNs, false, nil)
 	rules["required capabilities"] = topology.validateDSRequiredCapabilities()
 
+	//Get current Topology-CG for the requested change.
+	topoCachegroupNames := topology.getCachegroupNames()
+	userErr, sysErr, _ = dbhelpers.CheckTopologyOrgServerCGInDSCG(topology.ReqInfo.Tx.Tx, dsCDNs, topology.Name, topoCachegroupNames)
+	if userErr != nil {
+		return fmt.Errorf("%s", userErr)
+	}
+	if sysErr != nil {
+		log.Errorf("error while validate topology: %s", sysErr.Error())
```

Comment:
In addition to logging the sysErr, we should `return errors.New("unable to validate topology")`. We don't want to expose system errors to the user, so we return a generic error here instead (because this method only supports returning a single error currently).

Reply:
Understood.